### ISSUE TITLE
Testing with MCP30 printer

### ISSUE THREAD

**Issue by nneuberger1** [ovatu/starxpand_flutter#1]
I'm trying to configure this code with an MCP30 printer. It's typically used for a POS and it only has LAN connection according to this. I original bought this with a Square bundle package.

I ran the scan but nothing came back using the sample application.

The MCP30 apparently doesn't have bluetooth.

I originally started with the original iOS SDK test app.  I successfully printed from that app connected via both simulator and a physical device. I force entered my IP address for the printer and everything worked.

I'd like to just prototype the same with this app at first using the sample app.  Then I might be able to reverse engineer (or learn this code for the plugin) and potentially come up with a PR to support LAN based solutions.

I did see this starting point in SwiftStarxpandPlugin.swift

`
      let connection = StarConnectionSettings(interfaceType: InterfaceType.fromString(printer["interface"] as! String),
                                                            identifier: printer["identifier"] as! String, autoSwitchInterface: true)
`

It's similar to the original code from the iOS SDK Sample in PrintViewController (of that open source project).

` let starConnectionSettings = StarConnectionSettings(interfaceType: selectedInterface,
                                                            identifier: identifier)`

With that app, I entered into the screen my IP address.  I think I'm on the right track to see if I can just get connected to an IP address based printer.

I wasn't sure if this library was typically only used for just bluetooth printers exclusively.

It could perhaps be a routing issue why I don't discover my printer on the LAN. I'm trying to get the LAN routes completely open for both wireless and lan connected devices.

Let me know if I'm on the right track on getting an IP based printer working.

Thanks for sharing your code. I may just cut my loss with this MCP30 and find a bluetooth enabled version or some of the portable versions. If you have suggestions for purchase, that would be great.  Thanks again!

**Comment by nneuberger1** [comment]
Here's some MCP30 documentation so it appears that it only supports USB and LAN and I'll be using for both an iPad (with likely a Square stand) and mobile devices.  So likely it might be better to get bluetooth enabled devices
https://www.star-m.jp/products/s_print/mcprint3/manual/en/description/confirmedModels.htm

**Comment by nneuberger1** [comment]
I got it to work with relatively little test code!

I created a method like the following in main.dart of the sample app.

    _pressedLANPrint() async {
        var printer = StarXpandPrinter(model: StarXpandPrinterModel.mCPrint3, identifier: '192.168.10.251', interface: StarXpandInterface.lan);
    
    // call the existing method.
    await _print(printer);
    }

Then I added a button in the build method.

          ElevatedButton(child: Text('LAN Print'), onPressed: _pressedLANPrint),

At least the initial testing prototype is working! Out printed the ovatu receipt and the drawer opened!

**Comment by davidhole** [comment]
@nneuberger1 great that you're having success :)

Regarding the app not finding your LAN printer, this should definitely work, I would perhaps ensure that you have added the relevant keys for network access: NSLocalNetworkUsageDescription

Also, is the main star sample app able to find the printer and connect without you entering the IP address directly?

**Comment by nneuberger1** [comment]
I got it to work now. It turns out that I had a network route vlan configuration issue. your sample app now displays the printer and it prints fine. The original StarXpandSDK iOS app scans fine as well. I did notice the simulator via Xcode won't display the printer. That could be normal however. Let me know if that's expected.

**Comment by nneuberger1** [comment]
@davidhole can you recommend any scanners for a POS and will work with this library?

I didn't find much yet for API integration and some of the code to support it.

Thanks again for all of your help!

I purchased some cheaper USB scanners but not sure it's going to work with this library.

**Comment by davidhole** [comment]
@nneuberger1 this library wraps the StarXpand SDK so my understanding is that it only works with the mPOP scanner: https://star-emea.com/products/mpop/

This is what we use it with, it's not a general purpose barcode scanning SDK.

Not sure about the simulator, but I think that this is expected functionality, for POS printing, we test on device.

**Comment by nneuberger1** [comment]
@davidhole thanks for all your help. I ended up purchasing a socket mobile S740 device. Some of the specs look great and they have an API for various platforms including flutter.

https://store.socketmobile.com/products/socketscan-s740-universal-barcode-scanner

I've tried using much cheaper commodity devices like a Teva USB / Bluetooth scanner but I ran into so many issues with doing keyboard workarounds because most of them use some type of Keyboard HID mode. So it'll scan and receive the data, but the data is not correct. I tried search and replace functions to get the data back to the way it was, but iOS still sees it as a keyboard and thus blocking data entry.

This was the library I tried, but just didn't work well because of the Keyboard HID mode.
https://github.com/shaxxx/flutter_barcode_listener

Thanks again for your help.
